script.rb

```diff
@@ -0,0 +1,81 @@
+module Enumerable
+  def my_each(array)
+    index = 0
+    while index < array.length
+      yield(array[index])
+      index += 1
+    end
+    array
+  end
+
+  def my_each_with_index(array)
+    index = 0
+    while index < array.length
+      yield(array[index], index)
+      index += 1
+    end
+    array
+  end
+
+  def my_select(array)
+    new_arr = []
+    my_each(array) do |x|
+      new_arr.push(x) if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_all?(array)
+    p my_select(array) { |x| yield(x) }.length == array.length
+  end
+
+  def my_any?(array)
+    p my_select(array) { |x| yield(x) }.length.positive? ? true : false
+  end
+
+  def my_none?(array)
+    p my_select(array) { |x| yield(x) }.length.length.zero? ? true : false
+  end
+
+  def my_count(array, idx = 0)
+    index = idx
+
+    if block_given? == true
+      my_select(array) { |x| yield(x) }.length
+
+    else
+      array.length - index
+    end
+  end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_count
> Remember to test it not only for arrays but for ranges as well

- [ ] my_count returns the number of items in enum through enumeration
```ruby
[5, 4, 1].my_count == [5, 4, 1].count => 3
```
- Yours is giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/77696262-b4e3bc00-6f72-11ea-9cbe-75dc178d2de8.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696292-c200ab00-6f72-11ea-8e97-5a2b79256cbc.png)

- [ ] my_count counts the number of items in enum that are equal to the item if an argument is given
```ruby
[5, 4, 1].my_count(4) == [5, 4, 1].count(4) => 1
```
- Yours is giving an NoMethodError
![image](https://user-images.githubusercontent.com/52209067/77696347-d6dd3e80-6f72-11ea-8b7e-60078c7c0085.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696378-e2306a00-6f72-11ea-9310-0bb30ab9c94d.png)

- [ ] my_count counts the number of elements yielding a true value if a block is given
```ruby
[5, 4, 1].my_count { |x| x > 2 } == [5, 4, 1].count { |x| x > 2 }
```
- Yours is giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/77696427-faa08480-6f72-11ea-8113-b3fbc2fd1a81.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696440-055b1980-6f73-11ea-9b41-09dac295d3d3.png)